main/preload.ts

```diff
@@ -0,0 +1,74 @@
+import { contextBridge, ipcRenderer } from 'electron';
+import { ProgressState } from '../main/models/progressState.model';
+import { BKProject } from '../main/models/projectFormat.model';
+import { OpenDialogOptions, SaveDialogOptions } from '../src/App/components/file-dialog.model';
+import { RootDirectories } from '../src/models/store.model';
+import { SubmissionArgs, SubmissionReturn } from '../src/models/submission.model';
+
+export const api = {
+  /**
+   * Here you can expose functions to the renderer process
+   * so they can interact with the main (electron) side
+   * without security problems.
+   *
+   * The functions below can accessed using `window.api.fnName`
+   */
+
+  getBKProject: (rootDirectories: RootDirectories): void => {
+    ipcRenderer.send('did-start-getbkproject', rootDirectories);
+  },
+
+  onBKProject: (callback: Function): void => {
+    ipcRenderer.on('did-finish-getbkproject', (_event: Event, projects: BKProject) => {
```

Comment:
Its just one `project` (rather than plural `projects`).

Reply:
It should actually returns an array. The type here was wrong.